### PR TITLE
fixes #32 and fixes #51

### LINKED ISSUE #32: add filter for content_url and plugins_url
Because `content_url()` and `plugins_url()` are still based on some default constants (see link-template.php [L#2798](https://core.trac.wordpress.org/browser/tags/4.1/src/wp-includes/link-template.php#L2798) and [L#2830](https://core.trac.wordpress.org/browser/tags/4.1/src/wp-includes/link-template.php#L2830)), we have to add your `mangle_url` filtert here also.

I think [this](https://github.com/humanmade/Mercator/blob/master/mercator.php#L289) would be a nice place.

### LINKED ISSUE #51: Localize forgot password link
If you are www.site1.com and you click on the forgotten password link, it takes you to www.mainsite.com/wp-login.php?action=lostpassword . This is confusing, as this url as not the same as the one they came from and the users may think that this url is an error. In my tests, if go to www.site1.com/wp-login.php?action=lostpassword it seems to work. 

This *should* be as simple as this 

```php
add_filter( 'lostpassword_url', __NAMESPACE__ . '\\mangle_url', -10, 4 );
```

### PR DESCRIPTION
**Pull request by scarstens**

### CODE REVIEW

File: mercator.php [hunk 1]
```diff
@@ -324,3 +326,29 @@ function mangle_url( $url, $path, $orig_scheme, $site_id = 0 ) {
...
 }
+
+/**
+ * Mangle the content URL to give our mapped domain
+ *
+ * @param string $url The complete home URL including scheme and path.
+ * @param string $path Path relative to the home URL. Blank string if no path is specified.
+ * @return string Mangled Content URL
+ */
+function mangle_content_url( $url, $path) {
```
Comment (rmccue) [inline mercator.php]: Needs an extra space after `$path`

File: mercator.php [hunk 2]
```diff
@@ -324,3 +326,29 @@ function mangle_url( $url, $path, $orig_scheme, $site_id = 0 ) {
...
+	//get site ID for use with getting mapped domain name
```
Comment (rmccue) [inline mercator.php]: Comments need a space after the `//`, and capital letter to start

Comment (rmccue) [inline mercator.php]: Also, I'd drop this comment and the next (`pull in site aliases`) as they're not really adding much on top of the code itself here.

File: mercator.php [hunk 3]
```diff
@@ -324,3 +326,29 @@ function mangle_url( $url, $path, $orig_scheme, $site_id = 0 ) {
...
+	if ( empty( $current_mapping ) || $site_id !== (int) $current_mapping->get_site_id() ) {
+		return $url;
+	}
+
+	// Replace the domain with mapped domain within the url
+	$domain = parse_url( $url, PHP_URL_HOST );
+	$regex = '#^(\w+://)' . preg_quote( $domain, '#' ) . '#i';
+	$mangled = preg_replace( $regex, '${1}' . $current_mapping->get_domain(), $url );
+	
+	//return filtered content url
```
Comment (rmccue) [inline mercator.php]: I'd drop this comment too, I think `return $mangled` is self-evident already, and we don't want to overcomment :)

File: mercator.php [hunk 4]
```diff
@@ -324,3 +326,29 @@ function mangle_url( $url, $path, $orig_scheme, $site_id = 0 ) {
...
+ * @param string $url The complete home URL including scheme and path.
+ * @param string $path Path relative to the home URL. Blank string if no path is specified.
+ * @return string Mangled Content URL
+ */
+function mangle_content_url( $url, $path) {
+	//get site ID for use with getting mapped domain name
+	$site_id = get_current_blog_id();
+	//pull in site aliases
+	$current_mapping = $GLOBALS['mercator_current_mapping'];
+	//make sure the current domain is a mapped domain, else return the original url
```
Comment (rmccue) [inline mercator.php]: Reword? Something like `If mapping isn't valid for this site, return the original URL`

**Comment by rmccue** [comment]
Sorry for the overcommenting here, kind of stream-of-consciousness. Thanks for the PR!

Do we need the `mangle_content_url` function separately to the existing `mangle_url` function? Looks like both take the same first two parameters and return the same thing.

**Comment by scarstens** [comment]
I swear I tried using that function first, and there was some reason it didn't work with the other filters. I do agree reading it out of context here that it appears they should be able to merge IF we make the 3rd param optional. Oh, that was it, content_url filter only passes 2 params and causes an error when you try to use this function to filter. I can just set it to null by default. I'll try that.

**Comment by scarstens** [comment]
Yup, you were right, I removed the function, corrected everything and tested it.

**Pull request closed without merging**